htcanalyze/main.py

```diff
@@ -44,18 +51,18 @@
                          "errors", "ram-history"]
 
 
-# class to store and change global variables
-class GlobalPlayer(object):
+# this class sets the output and input streams as global variables
```

Comment:
Nitpick: we know it's a class. So "this class" can be removed.

Reply:
Good comment, now put it in the docstring instead :)

Reply:
Hm actually not that easy, because sys.stdin and sys.stdout can change at runtime, because of for ex. the ram-histogram created by plotille and I think even by the progress bars of rich.

Thats why I am saving it after start.

Reply:
I still don't think the class is necessary, although if this is the case you may want to include the `sys` results in the `run` function.

Currently you instantiate `GlobalPlayer`, setting its instance attributes to `None`. From `main()` you then issue `run()`, which is done only once. Inside of `run()` you call `GlobalPlayer.reset()`, setting the values to `None` again. This is redundant, because before this point you don't change `GlobalPlayer` (there is no other entry point to the script that would do this, and no other script imports `GlobalPlayer`). Next, you call `GlobalPlayer.check_for_redirection()`, which uses `sys` to set the instance attributes of `GlobalPlayer`.

If it is absolutely relevant to catch the state of `stdin` and `stdout` (which I'm not fully convinced of yet), it would be most relevant either at the start of the script (in the local scope), or in `main()` (name in the local scope) and passed along to `run()`. This will be functionally identical to how you're doing it, but requires no class.

```
def main():
    # Explain what you're doing here
    redirect_stdout = not sys.stdout.isatty()
    reading_stdin = not sys.stdin.isatty()
    stdin_input = sys.stdin.readlines() if reading_stdin else None 

    run(sys.argv[1:], redirect_stdout, reading_stdin, stdin_input)


def run(cargs, redirect_stdout, reading_stdin, stdin_input):
    ...
```

On a sidenote, I also suggest moving `run()` to a different file, so that `main()` only contains code pertaining to the argparse and entry point redirection. `run` should then be more appropriately named to what it's actually doing.